### ISSUE TITLE
Cannot select default method by fully qualified method name

### ISSUE THREAD

**Issue by akozlova** [junit-team/junit5#516]
Given the code:

```java
interface ATest {
   @Test
    default void testSmthCommon() {}
}

class BTest implements ATest {}
```
I expect that passing `BTest#testSmthCommon` as the _fully qualified method name_ would discover test method in `BTest` class. 

Instead, the JUnit Platform discovers method in the interface and then the `JupiterTestEngine` fails to run it.

I believe the problem is due to the use of `method.getDeclaringClass()` in `DiscoverySelectors#selectJavaMethod(String)`.

This was noticed in the latest `SNAPSHOT` version.

**Comment by sbrannen** [comment]
Thanks for raising the issue @akozlova!

I have confirmed that this is a bug.

**Comment by sbrannen** [comment]
I have also reworded the title accordingly.

**Comment by sbrannen** [comment]
FYI: I have introduced a failing test that reproduces this issue. See c8a589947f8073a00a35044dbb2c1138480a1f58.

**Comment by sbrannen** [comment]
Tentatively assigned to @bechte in conjunction with his work on #489.

**Comment by bechte** [comment]
This issue makes the difference of `JavaMethodSelector` and `JavaMethodSource` very obvious. A selector references to the _Thing Under Test_ (also requiring the explicit `Class` to run the tests with), while the source points to the place _where_ the code is defined. Therefore, for the selector we have to extract the class from the fully qualified name and cannot use the `getDeclaringClass()`. For the source, we really want to use `getDeclaringClass()` for its purpose.

I changed this accordingly in both classes (selector and source), see commit #a7c5808. This will be merged with #489 as it is part of the overall rework on this topic.

**Comment by bechte** [comment]
Merged into master.
